product_logistics_uom/models/product_template.py

```diff
@@ -0,0 +1,161 @@
+# Copyright 2020 Akretion (https://www.akretion.com).
+# @author Raphaël Reverdy <raphael.reverdy@akretion.com>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class ProductTemplate(models.Model):
+    _inherit = "product.template"
+
+    product_volume = fields.Float(
+        "Volume in product UOM",
+        compute="_compute_product_volume",
+        inverse="_inverse_product_volume",
+        digits="Volume",
+    )
+    product_weight = fields.Float(
+        "Weight in product UOM",
+        compute="_compute_product_weight",
+        digits="Stock Weight",
+        inverse="_inverse_product_weight",
+        store=True,
+    )
+
+    # remove rounding from volume and weight
+    # this is needed to avoid rounding errors when converting between units
+    # and is safe since we display the volume and weight in the product's
+    # volume and weight UOM. In the same time, we need to keep the volume
+    # we ensure that no information is lost by storing the volume and weight
+    # without rounding.
+    volume = fields.Float(digits=False)
```

Comment:
sorry for make noise, you can keep that code or just replace False by None, because we are override this attribute from odoo codebase https://github.com/odoo/odoo/blob/fa90f11227591ed185a091092aa913080b73e222/addons/product/models/product_template.py#L105